appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardContext.java

```diff
@@ -189,6 +190,7 @@
 import org.apache.naming.resources.Resource;
 import org.apache.naming.resources.WARDirContext;
 import org.apache.naming.resources.WebDirContext;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
```

Comment:
Not necessarily a complaint, but rather a concern: From a dependency management aspect: Does it make sense to make this core component of the server depend on one of the MicroProfile specs?

Reply:
Could a solution where a whitelist-type of file with a list of "skippable" interfaces/annotations that is parsed be a good replacement? Apologies if this doesn't make sense since I'm not that familiar with all of the server's dependency management tools

Reply:
Discussed with @aubi and we agree that yes pulling in the MicroProfile API dependency isn't ideal, the actual workaround for this would be quite a bit of effort. It would involve essentially as you say making a whitelist/blacklist extension point (e.g. via service loader or similar pattern) which we could use to cull servlets we don't want to initialise.